src/openrct2/core/String.cpp

```diff
@@ -82,8 +83,17 @@ namespace String
         WideCharToMultiByte(CODE_PAGE::CP_UTF8, 0, src.data(), srcLen, result.data(), sizeReq, nullptr, nullptr);
         return result;
 #else
+// Which constructor to use depends on the size of wchar_t...
+// UTF-32 is the default on most POSIX systems; Windows uses UTF-16.
+// Unfortunately, we'll have to help the compiler here.
+#if U_SIZEOF_WCHAR_T==4
+        icu::UnicodeString str = icu::UnicodeString::fromUTF32((const UChar32*) src.data(), src.length());
+#elif U_SIZEOF_WCHAR_T==2
         std::wstring wstr = std::wstring(src);
-        icu::UnicodeString str = icu::UnicodeString((char16_t*) wstr.c_str());
+        icu::UnicodeString str = icu::UnicodeString((const wchar_t*) wstr.c_str());
+#else
+#error Unsupported U_SIZEOF_WCHAR_T size
+#endif
```

Comment:
I'm really dissatisfied with this block. We could just assume `U_SIZEOF_WCHAR_T==4` and be done with it, but if we ever want to use this code path on Windows, we'll need something like this…